config/webpack/partial/base.js

```diff
@@ -71,7 +71,9 @@ const base = ({name, target}) => compose(
   // config object.
   assoc('entry', {
     index: [
-      ...__DEV__ ? [require.resolve('source-map-support/register')] : [],
+      ...__DEV__ || target === 'node'
+        ? [require.resolve('source-map-support/register')]
```

Comment:
You can make a partial out of this actually. There's an `entry` function that will let you do what you want.

```js
import {entry} from 'webpack-partial';

const useIfDevOrNode = compose(
  assoc('devtool', 'source-map'),
  entry(([...before, first]) => {
    return [...before, require.resolve('source-map-support/register'), first];
  })
);
```